### PR TITLE
[WIP] Various fixes, mostly SameDiff/Nd4j

### PR DESCRIPTION
**Pull request by rnett**
Fixes #8031.
Should also fix #7676 (although some of the javadocs are rather brief).

Going to add add a few more before merging.

### CODE REVIEW

**Comment by rnett** [comment]
The added conv1d test fails because of #8085, the actual equality check passes.  It should pass as soon as that is fixed.

**Comment by rnett** [comment]
Also removed Conv1D TF import, as its not used (Tensorflow has no Conv1D op, uses transform and Conv2D).

**Comment by rnett** [comment]
Fixing #8102.
Fixing #8100.

**Comment by rnett** [comment]
For #8100, I didn't touch `ImportClassMapping` or `LegacyOpMapper`, but everything else has been updated.

**Comment by rnett** [comment]
Fixes #8102.
Fixes #8100.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/SameDiff.java [hunk 1]
```diff
@@ -3143,10 +3185,16 @@ public INDArray outputSingle(Map<String, INDArray> placeholders, String output)
                 placeholders, batch, requiredActivations, activeListeners, at);
     }
 
+    /**
+     * See {@link #one(String, DataType, int...)}
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/SameDiff.java]: For all of these "default arg" methods - mention what the default is.
So something like `As per {@link #one(String, DataType, int...)} with the datatype set according to the ND4J default floating point type ({@link ...})`

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/transforms/Pad.java [hunk 2]
```diff
@@ -55,6 +55,14 @@ public Pad(SameDiff sd, SDVariable in, SDVariable padding, Mode mode, double pad
         addTArgument(padValue);
     }
 
+    public Pad(INDArray in, INDArray padding, INDArray out, Mode mode, double padValue){
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/transforms/Pad.java]: Should probably have `@NonNull` on in and padding args (out can be null though)

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/ops/transforms/Transforms.java [hunk 3]
```diff
@@ -789,7 +793,7 @@ public static INDArray lessThanOrEqual(INDArray first, INDArray ndArray) {
      * @return
      */
     public static INDArray lessThanOrEqual(INDArray first, INDArray ndArray, boolean dup) {
-        return exec(new OldLessThanOrEqual(first, ndArray, Nd4j.createUninitialized(DataType.BOOL, first.shape(), first.ordering())));
+        return Nd4j.getExecutioner().exec(new LessThanOrEqual(first, ndArray, Nd4j.createUninitialized(DataType.BOOL, first.shape(), first.ordering())))[0];
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/ops/transforms/Transforms.java]: Not a big deal, more an FYI: We have `Nd4j.exec(...)` methods. I'm tending to use those whenever adding new op exec.

File: nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/autodiff/opvalidation/LayerOpValidation.java [hunk 4]
```diff
@@ -958,13 +964,78 @@ public void testConv1dBasic() {
...
+                                new double[]{1355.1515f, 1373.3333f, 1391.5153f}
+                        },
+                        new double[][]{
+                                new double[]{1536.9697f, 1555.1515f, 1573.3333f}
+                        },
+                        new double[][]{
+                                new double[]{1718.7878f, 1736.9697f, 1755.1515f}
+                        }
+                }
+        );
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/autodiff/opvalidation/LayerOpValidation.java]: Java supports a shorter sort of bracket notation for initializing multi-dimensional primitive arrays:
```
        INDArray expected = Nd4j.createFromArray(
                new double[][][]{
                        {{82.42424f, 100.60606f, 118.78788f}},
                        {{264.2424f, 282.4242f, 300.6061f}},
                        {{446.0606f, 464.2424f, 482.424f}},
                        {{627.8788f, 646.0606f, 664.2424f}},
                        {{809.6970f, 827.8788f, 846.0606f}},
                        {{991.5152f, 1009.69696f, 1027.8788f}},
                        {{1173.3333f, 1191.5152f, 1209.6970f}},
                        {{1355.1515f, 1373.3333f, 1391.5153f}},
                        {{1536.9697f, 1555.1515f, 1573.3333f}},
                        {{1718.7878f, 1736.9697f, 1755.1515f}}});
```


```

**Comment by AlexDBlack** [comment]
LayerOpValidaiton.testConv1dBasic is also failing here but passing on master

**Pull request merged**